### PR TITLE
Test on Python 3.13-dev

### PR DESCRIPTION
**Pull request by jarrodmillman**

### CODE REVIEW

**Comment by stefanv** [comment]
Looks like an import from typing.io that should come straight from typing.

**Comment by jarrodmillman** [comment]
Yes, I am not sure where it is coming from. I suspect it may be

**Comment by stefanv** [comment]
` from importlib_resources._py3 import`

**Comment by stefanv** [comment]
https://github.com/python/importlib_resources/blob/main/NEWS.rst#v521

**Comment by stefanv** [comment]
Technically, this should be a dependency requirement on `ics_vtimezones`, but probably fine to set it in our package.

**Comment by stefanv** [comment]
Uh oh, they have a version pin :joy:

```
    ics-vtimezones 2020.1 depends on importlib_resources<2.0 and >=1.4
```

**Comment by stefanv** [comment]
We had an issue with this before, but seems OK now — see https://github.com/scientific-python/yaml2ics/pull/89

**Comment by stefanv** [comment]
According to SPEC0, we can drop 3.8 and 3.9 if we want to.

**Comment by stefanv** [comment]
Feel free to revert that last commit if you want to keep 3.8, 3.9 support.

I couldn't find where to change the pyupdate language version; maybe not used for this project yet.

**Comment by jarrodmillman** [comment]
The only place we use this is pinned to Python 3.8 (netlify).

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu, macos, windows]
-        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12-dev"]
```
Comment (jarrodmillman) [inline .github/workflows/test.yml:22]: We should probably keep 3.8 support for netlify.

Comment (stefanv) [reply .github/workflows/test.yml:22]: This is becoming quite a nuisance. Can we use a venv to build?

```
brew install python3
PYTHON=$(brew --prefix python)/libexec/bin/python
$PYTHON -m venv .venv
source .venv/bin/activate
```

or similar?

**Comment by jarrodmillman** [comment]
> I couldn't find where to change the pyupdate language version; maybe not used for this project yet.
```
requires-python = ">=3.8"
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (jarrodmillman) [reply .github/workflows/test.yml:22]: It hasn't a problem for me yet. If it does, I will look into it. For now, I will just revert it.

**Pull request merged**